lib/net/server/server.ts

```diff
@@ -12,22 +24,26 @@ const defaultOptions = {
  */
 export class Server extends EventEmitter {
   public config: Config
-
-  public key: Buffer | string = ''
-  public bootnodes: any | string = ''
+  public key: Buffer | undefined
+  public bootnodes: Bootnode[] = []
 
   protected refreshInterval: number
   protected protocols: Set<Protocol>
 
   public started: boolean
 
-  constructor(options: any) {
+  /**
+   * Create new server
+   * @param {ServerOptions}
+   */
+  constructor(options: ServerOptions) {
     super()
 
-    options = { ...defaultOptions, ...options }
-    this.config = options.config
+    this.config = options.config ?? new Config()
```

Comment:
I made this a conscious decision after this singleton discussion with @rumkin and others to not do a default instantiation of `Config` within the classes - I remember that I took some significant extra round on adopting the tests after that - since this would introduce the danger to have an inconsistent config state in the code base if passing in the central config object would be forgotten (which would not be detected with this changed setup).

I would suggest that we stick with that - at least if you just reintroduced for some general convenience - or did you have explicit reasons to introduce this here?

Will not make this a blocker here though due to the size of the PR, think it is better if we got this quickly merged after the latest problems with conflicting work states towards such big code changes. But we might want to reverse that later on.